privacyidea/lib/monitoringmodules/sqlstats.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+#
+#  2018-11-22 Initial create
+#             Cornelius Kölbel <cornelius.koelbel@netknights.it>
+#
+#  License:  AGPLv3
+#  contact:  http://www.privacyidea.org
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+__doc__ = """This module writes statistics data to the SQL database table "monitoringstats".
+"""
+import logging
+from privacyidea.lib.monitoringmodules.base import Monitoring as MonitoringBase
+from privacyidea.lib.pooling import get_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+from privacyidea.lib.utils import censor_connect_string, convert_timestamp_to_utc
+from privacyidea.lib.lifecycle import register_finalizer
+log = logging.getLogger(__name__)
+from sqlalchemy import MetaData, cast, String
+from sqlalchemy import asc, desc, and_, or_
+from privacyidea.models import MonitoringStats
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+import traceback
+from privacyidea.lib.tokenclass import AUTH_DATE_FORMAT
+from dateutil.tz import tzlocal, tzutc
+
+
+metadata = MetaData()
+
+
+class Monitoring(MonitoringBase):
+
+    def __init__(self, config=None):
+        self.name = "sqlstats"
+        self.config = config or {}
+
+        # We can use "sqlstats" as the key because the SQLAudit connection
```

Comment:
The comments here are from the SQL Audit class code -- I think it would be better to either reformulate them, or just remove them.

Reply:
done in fa76b0b

Reply:
:+1: